lib/components/viewers/route-viewer.tsx

```diff
@@ -18,33 +17,45 @@ import {
   getSortedFilteredRoutes
 } from '../../util/state'
 import { getFormattedMode } from '../../util/i18n'
+import {
+  SetViewedRouteHandler,
+  ViewedRouteObject,
+  ViewedRouteState
+} from '../util/types'
 import { StyledIconWrapper } from '../util/styledIcon'
 import PageTitle from '../util/page-title'
 
 import { RouteRow } from './RouteRow'
 
-class RouteViewer extends Component {
-  static propTypes = {
-    agencies: PropTypes.array,
-    filter: PropTypes.shape({
-      agency: PropTypes.string,
-      mode: PropTypes.string,
-      search: PropTypes.string
-    }),
-    findRouteIfNeeded: PropTypes.func,
-    hideBackButton: PropTypes.bool,
-    modes: PropTypes.array,
-    routes: PropTypes.array,
-    setViewedRoute: PropTypes.func,
-    transitOperators: PropTypes.array,
-    viewedRoute: PropTypes.shape({
-      patternId: PropTypes.string,
-      routeId: PropTypes.string
-    }),
-    // Routes have many more properties, but none are guaranteed
-    viewedRouteObject: PropTypes.shape({ id: PropTypes.string })
-  }
+interface FilterProps {
+  agency?: string
+  mode?: string
+  search?: string
+}
```

Comment:
Should this also be put in `types.ts`?

Reply:
It is not shared (yet). When that's the case we can move it.